### PR TITLE
Updated types for webextension-polyfill

### PR DESCRIPTION
**Pull request by Lusito**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [changes are generated from mozilla schema files](https://github.com/Lusito/webextension-polyfill-ts)
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

Other notes:

Took a while to figure out why the dprint would not work on webextension-polyfill (did not update since the introduction). I removed webextension-polyfill from dprint.jsonc and added dprint to my generator.

### CODE REVIEW

File: .dprint.jsonc [hunk 1]
```diff
@@ -28,7 +28,6 @@
         "./types/react-native",
         "./types/recurly__recurly-js",
         "./types/sawtooth-sdk",
-        "./types/webextension-polyfill",
```
Comment (sandersn) [inline .dprint.jsonc:31]: is this accurate? I still see at least one file that says it's generated automatically. Does the generation process now format with dprint?

**Review by sandersn — CHANGES_REQUESTED** [review]
I skimmed through the changes and left a comment on .dprint.jsonc. Realistically there are too many changes to be reviewed by a DT administrator. Is there anyone else associated with this project would could give the code changes a closer view?

If this is still being generated from another source, maybe there's no point in a line-by-line review, however. Is that the case? If so, why drop the dprint exception?

File: .dprint.jsonc (hunk 1 above)
Comment (mnoorenberghe) [reply .dprint.jsonc:31]: Yes, the generation process now uses dprint: https://github.com/Lusito/webextension-polyfill-ts/blob/3f2243a599988f93fd83159a6d54fb6e8f41baac/package.json#L25

**Review by mnoorenberghe — APPROVED** [review]

**Review by sandersn — APPROVED** [review]

**Comment by Lusito** [comment]
Ready to merge

**Comment by Lusito** [comment]
Thanks for the reviews and sorry for the delay, I've been out sick.

**Pull request merged**